### ISSUE TITLE
Location for storageState.json??

### ISSUE THREAD

**Issue by lakshbhutani** [microsoft/playwright#18939]
I have a micro front-end architecture for my react app. One micro front end(container) takes care of login page, I want to write tests related to login in that repo itself. But other micro front-end have dashboard and other pages(listing, etc) which requires user to be logged in.

So how can I write tests related to dashboard and other pages which requires authentication state(user to be logged in) but I want to use the logged in state of container?

Also, where does playwright store storageState.json as I want to access state this in all micro front ends for logged in routes?

**Comment by ValorHeart** [comment]
Hi, You need to save the current session storage state by using:
`await requestContext.storageState({ path: "${TestConfig.rootDir}/storageState.json" });`
Then you can load it using the configuration file
`config.use.storageState`

I think that the global setup feature might be very useful for you for that task: https://playwright.dev/docs/test-advanced#global-setup-and-teardown

**Comment by lakshbhutani** [comment]
Thanks @ValorHeart  for responding. I've done it in a similar way to run tests in a single repo.
But since it is a micro front-end architecture and login page is a part of container micro front-end and other pages are part of other micro front end's. 

So how can I store the auth state from one repository(micro frontend) config and share it with other micro front end's (which have different repositories.)
Basically is there a way to share storageState.json among different micro front-end's??

**Comment by ValorHeart** [comment]
@lakshbhutani , If your session doesn't expire you can save the storageState.json file and keep it in the required project, else you need somehow to start the webserver of your other project and run the global setup against that

**Comment by dgozman** [comment]
Closing because we lack the information to act on this issue. If you can provide a full repro, please file a new issue with it, and link to this one.
